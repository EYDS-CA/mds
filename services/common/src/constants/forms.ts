export enum FORM {
  // projects
  ADD_EDIT_PROJECT_SUMMARY = "ADD_EDIT_PROJECT_SUMMARY",
  ADD_MINE_MAJOR_APPLICATION = "ADD_MINE_MAJOR_APPLICATION",
  // reports
  VIEW_EDIT_REPORT = "VIEW_EDIT_REPORT",
  // permits
  REVOKE_DIGITAL_CREDENTIAL = "REVOKE_DIGITAL_CREDENTIAL",
  REQUEST_REPORT = "REQUEST_REPORT",
  ADD_REPORT_TO_PERMIT_CONDITION = "ADD_REPORT_TO_PERMIT_CONDITION",
  ADD_PERMIT_CONDITION_CATEGORY = "ADD_PERMIT_CONDITION_CATEGORY",
  EDIT_PERMIT_CONDITION = "EDIT_PERMIT_CONDITION",
  // tailings
  ADD_TAILINGS = "ADD_TAILINGS",
  // documents
  REPLACE_DOCUMENT = "REPLACE_DOCUMENT",
  DELETE_DOCUMENT = "DELETE_DOCUMENT",
  ARCHIVE_DOCUMENT = "ARCHIVE_DOCUMENT",
  EDIT_HELP_GUIDE = "EDIT_HELP_GUIDE",
  INLINE_EDIT_PERMIT_CONDITION_CATEGORY = "INLINE_EDIT_PERMIT_CONDITION_CATEGORY",
  PERMIT_CONDITION_REVIEW_ASSIGNMENT = "PERMIT_CONDITION_REVIEW_ASSIGNMENT",
  UPDATE_MAJOR_MINE_APPLICATION = "UPDATE_MAJOR_MINE_APPLICATION",
  UPDATE_IRT = "UPDATE_IRT"
}
