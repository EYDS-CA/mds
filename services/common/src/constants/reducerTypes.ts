// Misc
export const AUTHENTICATION = "AUTHENTICATION";
export const STATIC_CONTENT = "STATIC_CONTENT";
export const MODAL = "MODAL";
export const DOCUMENT_VIEWER = "DOCUMENT_VIEWER";
export const SYSTEM_FLAG = "SYSTEM_FLAG";

// Mines
export const MINES = "MINES";

// Parties
export const PARTIES = "PARTIES";

// OrgBook
export const ORGBOOK = "ORGBOOK";

// Compliance Orders
export const COMPLIANCE = "COMPLIANCE";

// MineSpace
export const MINESPACE = "MINESPACE";

// Permits
export const PERMITS = "PERMITS";

// Explosive Storage & Use Permits
export const EXPLOSIVES_PERMITS = "EXPLOSIVES_PERMITS";

// Search
export const SEARCH = "SEARCH";

// Variances
export const VARIANCES = "VARIANCES";

// Projects
export const PROJECTS = "PROJECTS";
export const PROJECT_SUMMARIES = "PROJECT_SUMMARIES";
export const INFORMATION_REQUIREMENTS_TABLE = "INFORMATION_REQUIREMENTS_TABLE";

// Core Users
export const USERS = "USERS";

// Incidents
export const INCIDENTS = "INCIDENTS";

// Work Information
export const MINE_WORK_INFORMATIONS = "MINE_WORK_INFORMATIONS";

// Reports
export const REPORTS = "REPORTS";

// Notices of Work
export const NOTICE_OF_WORK = "NOTICE_OF_WORK";

// Securities
export const SECURITIES = "SECURITIES";

// EPIC Information
export const EXTERNAL_AUTHS = "EXTERNAL_AUTHS";

// Notices of Departure
export const NOTICES_OF_DEPARTURE = "NOTICES_OF_DEPARTURE";

// Activities
export const ACTIVITIES = "ACTIVITIES";

// Tailings Storage Facilities
export const TAILINGS = "TAILINGS";

// Dams
export const DAMS = "DAMS";


//Verifiable Credentials
export const VERIFIABLE_CREDENTIALS = "VERIFIABLE_CREDENTIALS";
