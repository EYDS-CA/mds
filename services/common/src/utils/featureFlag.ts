import { detectProdEnvironment as IN_PROD } from "./environmentUtils";
import flagsmith from "flagsmith";

// Name of feature flags. These correspond to feature flags defined in flagsmith.
export enum Feature {
  MAJOR_PROJECT_ARCHIVE_FILE = "major_project_archive_file",
  DOCUMENTS_REPLACE_FILE = "major_project_replace_file",
  MAJOR_PROJECT_ALL_DOCUMENTS = "major_project_all_documents",
  MAJOR_PROJECT_DECISION_PACKAGE = "major_project_decision_package",
  FLAGSMITH = "flagsmith",
  TSF_V2 = "tsf_v2",
}

// Definition of legacy flags
// TODO: Remove after flagsmith is live
const Flags = {
  [Feature.MAJOR_PROJECT_ARCHIVE_FILE]: !IN_PROD(),
  [Feature.DOCUMENTS_REPLACE_FILE]: !IN_PROD(),
  [Feature.MAJOR_PROJECT_ALL_DOCUMENTS]: !IN_PROD(),
  [Feature.MAJOR_PROJECT_DECISION_PACKAGE]: !IN_PROD(),
  [Feature.TSF_V2]: !IN_PROD(),
  [Feature.FLAGSMITH]: !IN_PROD(),
};

export const initializeFlagsmith = async (flagsmithUrl, flagsmithKey) => {
  await flagsmith.init({
    api: flagsmithUrl,
    environmentID: flagsmithKey,
    cacheFlags: true,
    enableAnalytics: true,
  });
};

/**
 * Returns true if the given feature is enabled
 * @param feature Feature to verify
 * @returns true if the given feature is enabled
 */
export const isFeatureEnabled = (feature: Feature) => {
  if (Feature.FLAGSMITH in Flags && Flags[Feature.FLAGSMITH]) {
    return flagsmith.hasFeature(feature);
  }

  if (feature in Flags) {
    return Flags[feature];
  }

  return false;
};
