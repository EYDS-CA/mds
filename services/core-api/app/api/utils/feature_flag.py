from enum import Enum

from app.config import Config
from flagsmith import Flagsmith
from flask import current_app


class Feature(Enum):
    TSF_V2 = 'tsf_v2'
    MAJOR_PROJECT_REPLACE_FILE = 'major_project_replace_file'
    MINE_APPLICATION_FILE_UDPATE_ALERTS = 'mine_application_file_update_alerts'
    TRACTION_VERIFIABLE_CREDENTIALS = 'verifiable_credentials'
    #if enabled the credential offer will be the current development state of all the 2.0 changes, Q1 2024
    VC_ANONCREDS_20 = 'vc_mines_act_permit_20'   # pending anoncred content differences.
    VC_ANONCREDS_CORE = "vc_anoncreds_core"
    VC_ANONCREDS_MINESPACE = "vc_anoncreds_minespace"
    VC_W3C = "vc_w3c"
    CODE_REQUIRED_REPORTS = 'code_required-reports'
    PERMIT_DOCUMENT_KEYWORD_SEARCH = 'permit_document_keyword_search'
    AMS_AGENT = 'ams_agent'

    def __str__(self):
        return self.value


flagsmith = Flagsmith(
    environment_key=Config.FLAGSMITH_KEY,
    api_url=Config.FLAGSMITH_URL,
)


def is_feature_enabled(feature):
    try:
        feature = str(feature).strip()
        flags = flagsmith.get_environment_flags()

        return feature in flags.flags and flags.is_feature_enabled(feature)
    except Exception as e:
        current_app.logger.error(f'Failed to look up feature flag for: {feature}. ' + str(e))
        return False
