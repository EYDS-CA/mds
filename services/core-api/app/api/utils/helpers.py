import re


def clean_HTML_string(raw_html):
    cleanr = re.compile('<.*?>')
    cleantext = re.sub(cleanr, '', raw_html)
    return cleantext


"""
Add sufix to permit number adding one to versioning.
"""


def generate_permit_no_sufix(permit, separator, filling=2):
    result = permit.split(separator + "-")
    version_str = '0' if len(result) < 2 else result[1]
    version = str(int(version_str) + 1).zfill(filling)
    permit_no = separator + "-" + version

    return permit_no
