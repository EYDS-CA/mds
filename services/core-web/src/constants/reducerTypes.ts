// Documents
export const DOCUMENTS = "DOCUMENTS";

// Alerts
export const MINE_ALERTS = "MINE_ALERTS";