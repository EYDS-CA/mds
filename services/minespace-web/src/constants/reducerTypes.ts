// User Dashboard Info
export const USER_MINE_INFO = "USER_MINE_INFO";
