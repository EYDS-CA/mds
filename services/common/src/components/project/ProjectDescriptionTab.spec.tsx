import React from "react";
import { render } from "@testing-library/react";
import ProjectDescriptionTab from "./ProjectDescriptionTab";
import * as MOCK from "@mds/common/tests/mocks/dataMocks";
import { ReduxWrapper } from "@mds/common/tests/utils/ReduxWrapper";
import { PROJECTS, PERMITS } from "@mds/common/constants/reducerTypes";
import { BrowserRouter } from "react-router-dom";

const initialState = {
  [PROJECTS]: {
    project: {
      ...MOCK.PROJECT,
      project_summary: {
        ...MOCK.PROJECT.project_summary,
        authorizations: [...MOCK.PROJECT.project_summary.authorizations,
        {
          project_summary_authorization_guid: "c47161c0-8860-4125-8e68-fac3d98cf429",
          project_summary_guid: "8b4b9781-2e59-43ef-8164-4cc3b964417a",
          project_summary_permit_type: ["AMENDMENT"],
          project_summary_authorization_type: "REFUSE_DISCHARGE_PERMIT",
          existing_permits_authorizations: ['1234'],
          amendment_changes: null,
          amendment_severity: null,
          is_contaminated: null,
          new_type: "PER",
          authorization_description: "asdf",
          exemption_reason: "asdasdas",
          amendment_documents: [],
          exemption_requested: false,
          ams_tracking_number: "0",
          ams_outcome: "Some error",
          ams_status_code: "400",
          ams_submission_timestamp: "2024-05-24T19:17:09.212499+00:00",
        }
        ]
      }
    },
  },
  [PERMITS]: { permits: MOCK.PERMITS },
};

describe("ProjectDescriptionTab", () => {
  it("renders properly", () => {
    const { container } = render(
      <ReduxWrapper initialState={initialState}>
        <BrowserRouter>
          <ProjectDescriptionTab />
        </BrowserRouter>
      </ReduxWrapper>
    );
    expect(container).toMatchSnapshot();
  });
});
