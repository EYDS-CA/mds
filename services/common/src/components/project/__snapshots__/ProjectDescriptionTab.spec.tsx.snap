// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectDescriptionTab renders properly 1`] = `
<div>
  <div
    class="ant-row"
    style="margin-top: -8px; margin-bottom: -8px;"
  >
    <div
      class="ant-col ant-col-24"
      style="padding-top: 8px; padding-bottom: 8px;"
    >
      <div
        class="ant-row ant-row-space-between"
      >
        <div
          class="ant-col"
        >
          <h2
            class="ant-typography"
          >
            Project Description Overview
          </h2>
        </div>
        <div
          class="ant-col"
        >
          <button
            class="ant-btn ant-btn-primary"
            data-cy="view-project-description-details-button"
            type="button"
          >
            <span>
              View Project Description Details
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="ant-col ant-col-24"
      style="padding-top: 8px; padding-bottom: 8px;"
    >
      <div
        class="ant-typography"
      >
        Below are the authorization submissions and their status in the project description. Both the Major Mines Office and Ministry of Environments reviews must be completed for this stage to be considered complete.
      </div>
      <div
        class="bcgov-callout--warning bcgov-callout"
      >
        <p
          class="bcgov-callout-title"
        >
          Draft
        </p>
        <p>
          This project step has not been formally submitted by the proponent through MineSpace. MineSpace users can update text fields and add documents.
        </p>
      </div>
      <div
        class="bcgov-callout--danger bcgov-callout"
      >
        <p
          class="bcgov-callout-title"
        >
          Environmental Management Act Submission Unsuccessful
        </p>
        <p>
          One or more of your environment authorization applications has not been
                      submitted successfully.
        </p>
      </div>
      <h3
        class="ant-typography font-colour"
      >
        Submission Progress
      </h3>
      <h5
        class="ant-typography primary-colour"
      >
        Major Mines Office
      </h5>
      <span
        class="ant-typography desktop-bold"
      >
        Mines Act
      </span>
      <div
        class="ant-table-wrapper  core-table"
      >
        <div
          class="ant-spin-nested-loading"
        >
          <div
            class="ant-spin-container"
          >
            <div
              class="ant-table"
            >
              <div
                class="ant-table-container"
              >
                <div
                  class="ant-table-content"
                >
                  <table
                    style="table-layout: auto;"
                  >
                    <colgroup />
                    <thead
                      class="ant-table-thead"
                    >
                      <tr>
                        <th
                          class="ant-table-cell"
                        >
                          Type
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          Permit
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          Date
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          Status
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      class="ant-table-tbody"
                    >
                      <tr
                        class="ant-table-row ant-table-row-level-0 fade-in"
                        data-row-key="4d04995c-f90d-42fd-a951-60d9891f242c"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          <div
                            class="project_type-column"
                            title="Type"
                          >
                            <div />
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div
                            class="permit_no-column"
                            title="Permit"
                          >
                            N/A
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div
                            class="date_submitted-column"
                            title="Date"
                          >
                            May 24 2024, 12:17 (PDT)
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <span
                            class="ant-badge ant-badge-status ant-badge-not-a-wrapper"
                          >
                            <span
                              class="ant-badge-status-dot ant-badge-status-success"
                            />
                            <span
                              class="ant-badge-status-text"
                            >
                              Submitted
                            </span>
                          </span>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <br />
      <span
        class="ant-typography desktop-bold"
      >
        Forestry Act
      </span>
      <div
        class="ant-table-wrapper  core-table"
      >
        <div
          class="ant-spin-nested-loading"
        >
          <div
            class="ant-spin-container"
          >
            <div
              class="ant-table"
            >
              <div
                class="ant-table-container"
              >
                <div
                  class="ant-table-content"
                >
                  <table
                    style="table-layout: auto;"
                  >
                    <colgroup />
                    <thead
                      class="ant-table-thead"
                    >
                      <tr>
                        <th
                          class="ant-table-cell"
                        >
                          Type
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          Permit
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          Date
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          Status
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      class="ant-table-tbody"
                    >
                      <tr
                        class="ant-table-row ant-table-row-level-0 fade-in"
                        data-row-key="80fa60e3-0c8b-48e1-aede-ed2830e8ca42"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          <div
                            class="project_type-column"
                            title="Type"
                          >
                            <div />
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div
                            class="permit_no-column"
                            title="Permit"
                          >
                            N/A
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div
                            class="date_submitted-column"
                            title="Date"
                          >
                            May 24 2024, 12:17 (PDT)
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <span
                            class="ant-badge ant-badge-status ant-badge-not-a-wrapper"
                          >
                            <span
                              class="ant-badge-status-dot ant-badge-status-success"
                            />
                            <span
                              class="ant-badge-status-text"
                            >
                              Submitted
                            </span>
                          </span>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <br />
      <h5
        class="ant-typography primary-colour"
      >
        Ministry of Environment
      </h5>
      <span
        class="ant-typography desktop-bold"
      >
        Environmental Management Act
      </span>
      <div
        class="ant-typography"
      >
        An Environmental Protection Officer will contact you once your application is reviewed and accepted. In the meantime, to learn about the ministry’s structured application process and timelines to get a waste discharge authorization, please visit
         
        <a
          href="https://www2.gov.bc.ca/gov/content/environment/waste-management/waste-discharge-authorization/process"
          target="_blank"
        >
          The waste discharge authorization process
        </a>
        .
      </div>
      <div
        class="ant-alert ant-alert-error ant-alert-with-description"
        data-show="true"
        role="alert"
        style="margin-bottom: 12px;"
      >
        <span
          aria-label="close-circle"
          class="anticon anticon-close-circle ant-alert-icon"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="close-circle"
            fill="currentColor"
            fill-rule="evenodd"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M512 64c247.4 0 448 200.6 448 448S759.4 960 512 960 64 759.4 64 512 264.6 64 512 64zm0 76c-205.4 0-372 166.6-372 372s166.6 372 372 372 372-166.6 372-372-166.6-372-372-372zm128.01 198.83c.03 0 .05.01.09.06l45.02 45.01a.2.2 0 01.05.09.12.12 0 010 .07c0 .02-.01.04-.05.08L557.25 512l127.87 127.86a.27.27 0 01.05.06v.02a.12.12 0 010 .07c0 .03-.01.05-.05.09l-45.02 45.02a.2.2 0 01-.09.05.12.12 0 01-.07 0c-.02 0-.04-.01-.08-.05L512 557.25 384.14 685.12c-.04.04-.06.05-.08.05a.12.12 0 01-.07 0c-.03 0-.05-.01-.09-.05l-45.02-45.02a.2.2 0 01-.05-.09.12.12 0 010-.07c0-.02.01-.04.06-.08L466.75 512 338.88 384.14a.27.27 0 01-.05-.06l-.01-.02a.12.12 0 010-.07c0-.03.01-.05.05-.09l45.02-45.02a.2.2 0 01.09-.05.12.12 0 01.07 0c.02 0 .04.01.08.06L512 466.75l127.86-127.86c.04-.05.06-.06.08-.06a.12.12 0 01.07 0z"
            />
          </svg>
        </span>
        <div
          class="ant-alert-content"
        >
          <div
            class="ant-alert-message"
          >
            Submission Unsuccessful
          </div>
          <div
            class="ant-alert-description"
          >
            <div>
              <span
                class="ant-typography"
              >
                One or more of the environment authorization application was not submitted successfully. Please 
                <b>
                  retry the failed submission
                </b>
                 or 
                <b>
                  start a new application
                </b>
                 for the rejected authorization(s). You can link this submission to the new application on the Related Projects page.
              </span>
              <div
                class="ant-list ant-list-split project-description-tab-errors"
              >
                <div
                  class="ant-spin-nested-loading"
                >
                  <div
                    class="ant-spin-container"
                  >
                    <ul
                      class="ant-list-items"
                    >
                      <li>
                        <div
                          class="inline-flex"
                        >
                          <div
                            class="flex-4"
                          >
                            <div
                              class="ant-row"
                            >
                              <div
                                class="ant-col ant-col-21"
                              >
                                Refuse Discharge Permit 
        Amendment(1234): "Some error"
                              </div>
                            </div>
                          </div>
                        </div>
                      </li>
                    </ul>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-table-wrapper  core-table"
      >
        <div
          class="ant-spin-nested-loading"
        >
          <div
            class="ant-spin-container"
          >
            <div
              class="ant-table"
            >
              <div
                class="ant-table-container"
              >
                <div
                  class="ant-table-content"
                >
                  <table
                    style="table-layout: auto;"
                  >
                    <colgroup />
                    <thead
                      class="ant-table-thead"
                    >
                      <tr>
                        <th
                          class="ant-table-cell"
                        >
                          Type
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          Authorization
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          Tracking #
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          Date
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          Status
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      class="ant-table-tbody"
                    >
                      <tr
                        class="ant-table-row ant-table-row-level-0 fade-in"
                        data-row-key="c47161c0-8860-4125-8e68-fac3d98cf429"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          <div
                            class="project_type-column"
                            title="Type"
                          >
                            undefined - undefined
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div
                            class="permit_no-column"
                            title="Authorization"
                          >
                            1234
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div
                            class="ams_tracking_number-column"
                            title="Tracking #"
                          >
                            N/A
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div
                            class="date_submitted-column"
                            title="Date"
                          >
                            May 24 2024, 12:17 (PDT)
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <span
                            class="ant-badge ant-badge-status ant-badge-not-a-wrapper"
                          >
                            <span
                              class="ant-badge-status-dot ant-badge-status-error"
                            />
                            <span
                              class="ant-badge-status-text"
                            >
                              Rejected
                            </span>
                          </span>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
